inc/dashboard/namespace.php

```diff
@@ -0,0 +1,436 @@
+<?php
+/**
+ * Dashboard Analytics
+ *
+ * @package aws-analytics
+ */
+
+namespace Altis\Analytics\Dashboard;
+
+use function Altis\Analytics\Blocks\get_views as get_block_views;
+use function Altis\Analytics\Utils\query;
+
+/**
+ * Set up the Dashboard Analytics page.
+ */
+function setup() {
+	add_filter( 'manage_posts_columns', __NAMESPACE__ . '\\remove_default_columns', 10, 2 );
+	add_filter( 'post_row_actions', __NAMESPACE__ . '\\remove_post_row_actions', 10, 2 );
+	add_filter( 'manage_edit-xb_sortable_columns', __NAMESPACE__ . '\\xb_table_sorting' );
+	add_filter( 'request', __NAMESPACE__ . '\\xb_block_column_orderby' );
+	add_filter( 'bulk_actions-edit-xb', '__return_empty_array' );
+	add_filter( 'views_edit-xb', __NAMESPACE__ . '\\render_date_range_links' );
+	add_filter( 'months_dropdown_results', '__return_empty_array' );
+	add_filter( 'manage_xb_posts_columns', __NAMESPACE__ . '\\add_microcopy_to_column_titles', 99 );
+
+	add_action( 'pre_get_posts', __NAMESPACE__ . '\\modify_views_list_query' );
+	add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\\enqueue_styles' );
+}
+
+/**
+ * Enqueue styles.
+ */
+function enqueue_styles() {
+	wp_enqueue_style( 'xb-insights', plugin_dir_url( __FILE__ ) . '/assets/style.css', [], '18-03-2021', 'screen' );
+}
+
+/**
+ * Remove the default post list columns.
+ *
+ * @param array $columns An array of columns to filter.
+ * @param string $post_type The current post type.
+ *
+ * @return array The filtered columns array.
+ */
+function remove_default_columns( $columns, $post_type ) : array {
+	if ( 'xb' === $post_type ) {
+		unset( $columns['cb'] );
+		unset( $columns['title'] );
+		unset( $columns['altis_publication_checklist_status'] );
+	}
+
+	return $columns;
+}
+
+/**
+ * Remove the post quick actions.
+ *
+ * @param array $actions An array of actions that can be done on a single post.
+ * @param object $post The WP_Post object
+ *
+ * @return array The filtered array (empty for xbs).
+ */
+function remove_post_row_actions( $actions, $post ) : array {
+	if ( 'xb' === $post->post_type ) {
+		unset( $actions['edit'] );
+		unset( $actions['view'] );
+		unset( $actions['trash'] );
+		unset( $actions['inline hide-if-no-js'] );
+	}
+
+	return $actions;
+}
+
+/**
+ * Add custom sort by columns
+ *
+ * @param array $columns An array of post list columns.
+ *
+ * @return array The updated array with sortable columns.
+ */
+function xb_table_sorting( $columns ) : array {
+	$columns['block'] = 'block';
+	$columns['views'] = 'views';
+	$columns['conversion'] = 'conversion';
+	return $columns;
+}
+
+/**
+ * Add additional microcopy to column titles.
+ *
+ * Note: This adds an additional span around the column title as there is no current way to filter the column title span itself.
+ *
+ * @see https://github.com/WordPress/WordPress/blob/643888dfc6697877609672087b4d3b4e5bf545ff/wp-admin/includes/class-wp-posts-list-table.php#L1059
```

Comment:
That makes sense. I was concerned about linking directly to a file in, say, the `master` branch because the line number isn't guaranteed to be the same, and context is important. Theoretically, that could happen in a branch, too, but possibly less often? Anyway, linking to a specific commit at least means that the link/line I'm pointing to wouldn't ever change. That was the thinking anyway. `¯\_(ツ)_/¯`